File: mvp-java/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/search/SearchView.java
package com.wasisto.githubuserfinder.android.ui.search;

import com.wasisto.githubuserfinder.domain.models.SearchHistoryItem;
import com.wasisto.githubuserfinder.domain.models.SearchUsersResult;

import java.util.List;

public interface SearchView {

    void showLoadingIndicator();

    void hideLoadingIndicator();

    void setQuery(String query);

    void showSearchResult(List<SearchUsersResult.Item> searchResult);

    void showSearchHistory(List<SearchHistoryItem> searchHistory);

    void hideSearchHistory();

    void showNoResultsText();

    void hideNoResultsText();

    void openUserDetailsActivity(String username);

    void showErrorToast();
}

File: mvp-kotlin/ui/src/main/java/com/wasisto/githubuserfinder/android/ui/search/SearchView.kt
package com.wasisto.githubuserfinder.android.ui.search

import com.wasisto.githubuserfinder.domain.models.SearchHistoryItem
import com.wasisto.githubuserfinder.domain.models.SearchUsersResult

interface SearchView {

    fun showLoadingIndicator()

    fun hideLoadingIndicator()

    fun setQuery(query: String?)

    fun showSearchResult(searchResult: List<SearchUsersResult.Item>)

    fun showSearchHistory(searchHistory: List<SearchHistoryItem>)

    fun hideSearchHistory()

    fun showNoResultsText()

    fun hideNoResultsText()

    fun openUserDetailsActivity(username: String)

    fun showErrorToast()
}
